pandas/_libs/src/vendored/ujson/python/objToJSON.c

```diff
@@ -713,7 +713,22 @@ void PdBlock_iterBegin(JSOBJ _obj, JSONTypeContext *tc) {
         return;
     }
 
-    arrays = get_sub_attr(obj, "_mgr", "column_arrays");
+    NPY_DATETIMEUNIT dunit = ((PyObjectEncoder *)tc)->datetimeUnit;
+    char *date_unit;
+    if (dunit == NPY_FR_s) {
+        date_unit = "s";
+    } else if (dunit == NPY_FR_ms) {
+        date_unit = "ms";
+    } else if (dunit == NPY_FR_us) {
+        date_unit = "us";
+    } else if (dunit == NPY_FR_ns) {
+        date_unit = "ns";
+    }
+
+    PyObject *mgr = PyObject_GetAttrString(obj, "_mgr");
```

Comment:
There might be some code segments that hit this but don't have a _mgr attr. This would return NULL and could explain the segfaults

So you will want to check if the result is equal to NULL and return an appropriate error message

Reply:
will update, but we only get here with a DataFrame.